### PR TITLE
Add Rollup to project

### PR DESCRIPTION
**Pull request by fernando24164**
Hi,

I check #18 issue to add Rollup to the project.

If you think this it is useful for you, I would appreciate you will add hacketoberfest-accepted tag.

Thx!

### CODE REVIEW

**Comment by jamescallumyoung** [comment]
Hi @fernando24164, thanks for the contribution. This project is open-source and has opt-in for hacktoberfest since I'm a fan of the project.

However, contributors should read the CONTRIBUTING guide before contributing. This is good open source etiquette and following the processes defined there helps maintainers manage their projects. It also helps reduce wasted effort. In this case, you didn;t indicate your interest in the issue so we didn;t have a chance to discuss your proposed changes.

That said, I'm happy to approcve your contributing with a few changes.

Could you explain your reasoning for adding Babel and changing the import/export method to `* from` in some cases? It's my understanding that Rollup (using the plugin specified in #18 ) can handle TS without needing Babel.

File: package.json [hunk 1]
```diff
@@ -12,7 +12,8 @@
     "dist"
   ],
   "scripts": {
-    "build": "tsc",
+    "build": "tsc --allowSyntheticDefaultImports",
```
Comment (jamescallumyoung) [inline package.json:15]: Can you specify why the `--allowSyntheticDefaultImports` flag is needed here and not in `build:js`

File: package.json [hunk 2]
```diff
@@ -36,6 +37,13 @@
   },
   "homepage": "https://jych.gitlab.io/gif-provider/docs/",
   "devDependencies": {
+    "@babel/core": "^7.9.0",
+    "@babel/node": "^7.10.5",
+    "@babel/preset-env": "^7.9.5",
+    "@babel/preset-typescript": "^7.9.0",
```
Comment (jamescallumyoung) [inline package.json:43]: As mentioned in my main comment, please specify why Babel is needed.

File: src/index.ts [hunk 3]
```diff
@@ -1,8 +1,8 @@
 // The gif
-export {Gif} from './Gif'
+export * as Gif from './Gif'
```
Comment (jamescallumyoung) [inline src/index.ts:2]: Why the `* from` change?

File: src/providers/GiphyGifProvider.ts [hunk 4]
```diff
@@ -3,7 +3,7 @@
 import {Gif} from "../Gif";
 import {GifProvider} from "../GifProvider";
 
-import giphyApi = require("giphy-api");
+import giphyApi from "giphy-api";
```
Comment (jamescallumyoung) [inline src/providers/GiphyGifProvider.ts:6]: Why change from `require`?

File: src/providers/index.ts [hunk 5]
```diff
@@ -1,5 +1,5 @@
 // Add new GifProviders here to have them exposed by the module
 
-export {CompositeGifProvider} from './CompositeGifProvider'
-export {GiphyGifProvider} from './GiphyGifProvider';
-export {TenorGifProvider} from './TenorGifProvider'
+export {CompositeGifProvider as CompositeGifProvider} from './CompositeGifProvider'
+export {GiphyGifProvider as GiphyGifProvider} from './GiphyGifProvider';
+export {TenorGifProvider as TenorGifProvider} from './TenorGifProvider'
```
Comment (jamescallumyoung) [inline src/providers/index.ts:5]: Is this needed? Otherwise, please reverse.

File: package.json [hunk 6]
```diff
@@ -12,7 +12,8 @@
     "dist"
   ],
   "scripts": {
-    "build": "tsc",
+    "build": "tsc --allowSyntheticDefaultImports",
+    "build:js": "npm run build && rollup -c",
```
Comment (jamescallumyoung) [inline package.json:16]: This project uses `yarn` not `npm`. Please update to:
```suggestion
    "build:js": "yarn build && rollup -c",
```

File: .babelrc [hunk 7]
```diff
@@ -0,0 +1,7 @@
+{
+    "presets": [
+        "@babel/env",
+        "@babel/typescript"
+    ],
+    "plugins": []
+}
```
Comment (jamescallumyoung) [inline .babelrc:7]:
```suggestion
}

```

**Review by jamescallumyoung — CHANGES_REQUESTED** [review]

**Comment by fernando24164** [comment]
Sorry for not take a look to CONTRIBUTING file. I will update the issue and the PR ASAP.
Thx

**Review by fernando24164 — COMMENTED** [review]
I added the new changes to get updated with the last changes from master and I have also modified commit message to be properly defined like the rest of the project.
My first solution for resolve #18 was parsing typescript code using babel project. Reading more carefully your proposal typescript plugin resolve everything out of the box without babel.
The only problem I have found it is defined root directory in tsconfig due this plugin needs it to generate the final js file.
Finally, I added [terser](https://github.com/terser/terser) [plugin](https://github.com/TrySound/rollup-plugin-terser) to Rollup to try to compress as much as possible the final js file size.

**Comment by jamescallumyoung** [comment]
Stale.

**Pull request closed without merging**